app/components/navigation_header/search.tsx

```diff
@@ -52,18 +52,21 @@ const NavigationSearch = ({
         searchProps.onFocus?.(e);
     }, [hideHeader, searchProps.onFocus]);
 
-    useEffect(() => {
-        const show = Keyboard.addListener('keyboardDidShow', () => {
-            if (Platform.OS === 'android') {
-                DeviceEventEmitter.emit(Events.TAB_BAR_VISIBLE, false);
-            }
-        });
+    const showEmitter = useCallback(() => {
+        if (Platform.OS === 'android') {
+            DeviceEventEmitter.emit(Events.TAB_BAR_VISIBLE, false);
+        }
+    }, []);
 
-        const hide = Keyboard.addListener('keyboardDidHide', () => {
-            if (Platform.OS === 'android') {
-                DeviceEventEmitter.emit(Events.TAB_BAR_VISIBLE, true);
-            }
-        });
+    const hideEmitter = useCallback(() => {
+        if (Platform.OS === 'android') {
+            DeviceEventEmitter.emit(Events.TAB_BAR_VISIBLE, true);
+        }
+    }, []);
+
+    useEffect(() => {
+        const show = Keyboard.addListener('keyboardDidShow', showEmitter);
+        const hide = Keyboard.addListener('keyboardDidHide', hideEmitter);
```

Comment:
I am curious about this change. Both showEmitter and hideEmitter are only needed in the body of the effect. Moving then outside only adds legibility, but not passing them as dependencies to the effect may lead to future issues, since eventually you may add some dependency to the callback, and that dependency / change won't be reflected in the effect.

So, long story short:
- It is ok as it is
- 0/5 revert and keep it as before to reduce slightly the useCallback (almost null) performance and memory impact.
- 0/5 passing showEmitter and hideEmitter to the dependencies of the effect, to be future proof, at the cost of the (minimal) performance implication of having dependencies in the effect that will never trigger.

Up to you :smile:

Reply:
for that reason, I opted for your 3rd suggestion and added showEmitter and hideEmitter to the useEffect dependencies.